### PR TITLE
Datacite xml improvements

### LINKED ISSUE #5889: Align or merge DataCite metadata exports
This issue is meant to record the differences between Dataverse's two newest metadata exports as of v4.14, **"DataCite"/"Datacite"** and **"OpenAIRE"/"oai_datacite"**, and discussion about how to align (or possibly merge) the very similar exports.

As part of v4.10 (released in Dec. 2018), Dataverse makes available through the UI, API and over OAI-PMH dataset metadata in the DataCite schema (https://github.com/IQSS/dataverse/issues/5043). This lets Dataverse export dataset metadata in a widely-used, discipline-agnostic schema that's more standardized than Schema.org and has more metadata than Dublin Core.

As part of v4.14 (released in May 2019), Dataverse makes available through the UI, API and over OAI-PMH DataCite metadata that complies with OpenAIRE requirements (https://github.com/IQSS/dataverse/issues/4257). Repositories need to follow these requirements in order for their dataset metadata to be made discoverable (harvested) by OpenAIRE ([OpenAIRE EXPLORE](https://explore.openaire.eu/search/find)). The OpenAIRE metadata requirements follow the DataCite schema, with [some differences between OpenAIRE and DataCite listed in their documentation](https://guidelines.openaire.eu/en/latest/data/use_of_datacite.html).

What both exports are called depending on the export method:

openaire-datacite-export-graphic

Both metadata exports are based on DataCite 4 and are meant to be valid against the DataCite 4 schema (although the xml records available over OAI-PMH in "Datacite" format reference DataCite's 3.1 schema). But Dataverse exports them as separate formats for several reasons:
- The two metadata exports were worked on at different times by different groups
- When work on making Dataverse OpenAIRE compliant started, I thought the OpenAIRE export would follow the DataCite 3.1 schema since the [OpenAIRE guidelines for data repositories](https://guidelines.openaire.eu/en/latest/data/index.html) follows DataCite 3.1. And I knew that Dataverse would eventually export DataCite 4 metadata, so it made sense to make them separate exports. But we're told the OpenAIRE folks plan to update their guidelines, so our 4Science colleagues created the OpenAIRE export following the DataCite 4 schema. (For example, a notable difference between DataCite 3 and 4 is how funder information is handled. [The OpenAIRE guidelines mandate](https://guidelines.openaire.eu/en/latest/data/use_of_datacite.html#funding-information) that the contributorType property is used, which is how DataCite 3 handles funder info. But Dataverse's OpenAIRE export is using the DataCite 4 fundingReferences property instead.)
- The "OpenAIRE" metadata export uses an algorithm that adds metadata about whether dataset authors and contact persons are people or organizations (in DataCite's nameType attribute). The algorithm was [the last thing discussed in the OpenAIRE GitHub issue](https://github.com/IQSS/dataverse/issues/4257#issuecomment-483562638).

Ideally, Dataverse would export only one  metadata record, made available through the UI, API and over OAI-PMH, that follows the DataCite schema and is also OpenAIRE compliant. The way things are now, where Dataverse exports two different metadata records based on DataCite but different, we suspect people who want to export or harvest metadata will be confused about the differences between the two metadata export called "DataCite" and "OpenAIRE" in the UI and called "Datacite" and "oai_datacite" in the API endpoints and made available over OAI-PMH.

But we may want to maintain two metadata exports because:
- the OpenAIRE export is using the nameType algorithm, which was tested during QA but only tested for evidence that the algorithm would work in at least some cases. We haven't tried to estimate how often it will correctly figure out if author/contact names of actual datasets are people or organizations (although it's based on an algorithm DataCite uses that we're told is right over 90% of the time). Would people want to be able to export or harvest metadata that does not include the nameType metadata (maybe because they find that it's not correct often enough)?
- the OpenAIRE export uses [one of four mandatory Access Rights](https://wiki.surfnet.nl/display/standards/info-eu-repo#info-eu-repo-AccessRights) terms. The rules that Dataverse uses to determine this are discussed in [a GitHub issue comment](https://github.com/IQSS/dataverse/issues/4257#issuecomment-394409209). But I realized recently that the rules are too simple and lead to cases where datasets are marked as closedAccess when restricted access is more appropriate (e.g. https://doi.org/10.7910/DVN/0PMZC6, where file request is disabled, but people can request access through a process that happens outside of Dataverse). A GitHub issue about this is opened (https://github.com/IQSS/dataverse/issues/5920), so we can figure out how to assign more appropriate access rights to datasets. Until then, would people want to be able to export or harvest metadata that does not include these sometimes misleading Access Rights?

We should decide if:
- Dataverse should maintain one export or two and
- If maintaining only one export, make sure that it has all of the metadata available in the current two exports.
- If maintaining two exports, make sure that the amount of metadata in one export is as close to the same amount in the other (and continues to be as synced as possible) and document what the differences are. (As of v4.14 the "OpenAIRE" export has more metadata than the "DataCite" export but there are things missing in both.)

### PR DESCRIPTION
**Pull request by qqmyers**
Note - this is a subset of the changes in #10632 / #10632 adds relatedPublication relationType in addition to updating the XML sent to DataCite as described here. Nominally only #10632 needs to be merged, but this could be merged first to make it easier to see just the relationType changes in #10632.

**What this PR does / why we need it**: This PR updates the XML sent to DOI providers to be richer and more consistent with what is in the OpenAire export. There are differences (most obviously, the code here works for draft datasets and files which are not handled by the OpenAire exporter), most are probably subtle but @jggautier or others may wish to look at/comment on the list of changes I've tried to track in [this doc](https://docs.google.com/document/d/1JzDo9UOIy9dVvaHvtIbOI8tFU6bWdfDfuQvWWpC0tkA/edit?usp=sharing).

This is nominally one of 2 PRs related to DOI metadata - the other will be a change to the citation metadatablock to add a relationshipType to the field(s) where we expect to report the relationships to DataCite/etc.

**Which issue(s) this PR closes**:
Probably closes #5889 when combined with the relationshipType PR.
Closes #

**Special notes for your reviewer**: The PR includes some refactoring of our XML generation code that is common with the DDI export. @poikilotherm's idea of generating a model from the DataCite schema might be a good next step (hopefully easier given the code here to get the content needed). 

**Suggestions on how to test this**: QDR is testing this code now. There is one new test in addition to the existing ones - it could/should be expanded to test a dataset with more of the metadata that is covered by the new code - possibly just reading in one of the ~complete dataset examples we have.  Verification that the result matches the DataCite schema is included.

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**: only changes the XML sent to the DOI provider which is also exposed as a metadata export. The change makes the OpenAire export somewhat redundant, but as noted above, there are differences which may need to be kept. 

**Is there a release notes update needed for this change?**: yes - will create once the set of changes (in the doc above) is reviewed. Probably should be combined with the citation block change if that can be created in time for the same release.

**Additional documentation**:

### CODE REVIEW

**Comment by jggautier** [comment]
Thanks @qqmyers! I'll try to leave questions in the Google Doc this week or early next week.

**Comment by DS-INRA** [comment]
Another related issue (probably already taken into account since it's on the QDR board :) but just in case) :
- #2917

**Comment by pdurbin** [comment]
Just a heads up that I'm tasked with sending "Software" etc in another issue and I can only assume it will cause merge conflicts if we merge my PR first. Here's the commit: https://github.com/IQSS/dataverse/pull/10694/commits/8593d328ea5fbb94c1435b0ff4651199dcf7abef

**Comment by pdurbin** [comment]
@qqmyers can you please resolve merge conflicts?

File: src/main/java/edu/harvard/iq/dataverse/DataFile.java [hunk 1]
```diff
@@ -1123,4 +1123,23 @@ private boolean tagExists(String tagLabel) {
         }
```
Comment (pdurbin) [inline src/main/java/edu/harvard/iq/dataverse/DataFile.java:1123]: I'll just leave this comment here on the first file that is changed...

Can we please get a release note?

**Review by pdurbin — COMMENTED** [review]
Needs a release note

**Comment by qqmyers** [comment]
Closing in favor of #10632, which now has some bug fixes in the XML that haven't been back-ported to this PR.

**Pull request closed without merging**